File: evidence/generated-systems/angular/insurance-on-angular/src/app/models/Policy.js

// Define collection and schema for Policy
export interface Policy {
    policyNumber:
	type : PolicyNumber
    effectivePeriod:
	type : DateRange
    totalPremium:
	type : Money
    Insurer:
	type : Schema.Types.ObjectId
    Customer:
	type : Schema.Types.ObjectId
    Product:
	type : Schema.Types.ObjectId
    Agent:
	type : Schema.Types.ObjectId
    Coverages:
 	type : [{ type: Schema.Types.ObjectId, ref: 'PolicyCoverage' }]
    InsuredObjects:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InsuredObject' }]
    Endorsements:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Endorsement' }]
    BillingAccount:
	type : Schema.Types.ObjectId
    Beneficiaries:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Beneficiary' }]
    Claims:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Claim' }]
    ReinsuranceAgreements:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ReinsuranceAgreement' }]
    Status:
 	type : String
    PaymentPlan:
 	type : String
#
    collection: 'policys'
}

File: evidence/generated-systems/django/insurance-on-django/insuranceOnDjango/models/Policy.py
from django.db import models
from insuranceOnDjango.models.PolicyStatus import PolicyStatus
from insuranceOnDjango.models.PaymentPlanType import PaymentPlanType

#======================================================================
# 
# Encapsulates data for model Policy
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Policy Declaration
#======================================================================
class Policy (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	policyNumber = PolicyNumber
	effectivePeriod = DateRange
	totalPremium = Money
	insurer = models.ForeignKey('Insurer', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	customer = models.ForeignKey('Customer', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	product = models.ForeignKey('InsuranceProduct', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	agent = models.ForeignKey('Agent', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	coverages = models.ManyToManyField('PolicyCoverage',  blank=True, related_name='+')
	insuredObjects = models.ManyToManyField('InsuredObject',  blank=True, related_name='+')
	endorsements = models.ManyToManyField('Endorsement',  blank=True, related_name='+')
	billingAccount = models.ForeignKey('BillingAccount', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	beneficiaries = models.ManyToManyField('Beneficiary',  blank=True, related_name='+')
	claims = models.ManyToManyField('Claim',  blank=True, related_name='+')
	reinsuranceAgreements = models.ManyToManyField('ReinsuranceAgreement',  blank=True, related_name='+')
	status = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in PolicyStatus])
	paymentPlan = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in PaymentPlanType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.policyNumber
		str = str + self.effectivePeriod
		str = str + self.totalPremium
		str = str + self.status
		str = str + self.paymentPlan
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Policy";
    
	def objectType(self):
		return "Policy";
